pkg/routeagent_driver/handlers/ovn/submarinerNonGWRouteHandler.go

```diff
@@ -0,0 +1,172 @@
+	if !ok {
+		return errors.Wrapf(err, "error parsing the transit switch IP")
+	}
+
+	return nil
+}
+
+func (nonGatewayRouteHandler *NonGatewayRouteHandler) GetName() string {
+	return "submariner-nongw-route-handler"
+}
+
+func (nonGatewayRouteHandler *NonGatewayRouteHandler) GetNetworkPlugins() []string {
+	// TODO enable when we switch to new implementation
+	// return []string{cni.OVNKubernetes}
+	return []string{}
+}
+
+func (nonGatewayRouteHandler *NonGatewayRouteHandler) RemoteEndpointCreated(endpoint *submarinerv1.Endpoint) error {
+	nonGatewayRouteHandler.mutex.Lock()
+	defer nonGatewayRouteHandler.mutex.Unlock()
+
+	nonGatewayRouteHandler.remoteEndpoints[endpoint.Name] = endpoint
+
+	if !nonGatewayRouteHandler.isGateway || nonGatewayRouteHandler.transitSwitchIP == "" {
+		return nil
+	}
+
+	_, err := nonGatewayRouteHandler.smClient.SubmarinerV1().
+		NonGatewayRoutes(endpoint.Namespace).Create(context.TODO(),
+		nonGatewayRouteHandler.newNonGatewayRoute(endpoint), metav1.CreateOptions{})
+	if !apierrors.IsAlreadyExists(err) {
+		return errors.Wrapf(err, "error processing the remote endpoint create event for %q", endpoint.Name)
+	}
+
+	return nil
+}
+
+func (nonGatewayRouteHandler *NonGatewayRouteHandler) RemoteEndpointRemoved(endpoint *submarinerv1.Endpoint) error {
+	nonGatewayRouteHandler.mutex.Lock()
+	defer nonGatewayRouteHandler.mutex.Unlock()
+	delete(nonGatewayRouteHandler.remoteEndpoints, endpoint.Name)
+
+	if !nonGatewayRouteHandler.isGateway || nonGatewayRouteHandler.transitSwitchIP == "" {
+		return nil
+	}
+
+	if err := nonGatewayRouteHandler.smClient.SubmarinerV1().NonGatewayRoutes(endpoint.Namespace).Delete(context.TODO(),
+		endpoint.Name, metav1.DeleteOptions{}); err != nil {
+		return errors.Wrapf(err, "error deleting nonGatewayRoute %q", endpoint.Name)
+	}
+
+	return nil
+}
+
+func (nonGatewayRouteHandler *NonGatewayRouteHandler) TransitionToNonGateway() error {
+	nonGatewayRouteHandler.mutex.Lock()
+	defer nonGatewayRouteHandler.mutex.Unlock()
+
+	nonGatewayRouteHandler.isGateway = false
+
+	return nil
+}
+
+func (nonGatewayRouteHandler *NonGatewayRouteHandler) TransitionToGateway() error {
+	nonGatewayRouteHandler.mutex.Lock()
+	defer nonGatewayRouteHandler.mutex.Unlock()
+
+	nonGatewayRouteHandler.isGateway = true
+	if nonGatewayRouteHandler.transitSwitchIP != "" {
+		for _, endpoint := range nonGatewayRouteHandler.remoteEndpoints {
+			if _, err := nonGatewayRouteHandler.smClient.SubmarinerV1().NonGatewayRoutes(endpoint.Namespace).Update(context.TODO(),
+				nonGatewayRouteHandler.newNonGatewayRoute(endpoint), metav1.UpdateOptions{}); err != nil {
+				if !apierrors.IsNotFound(err) {
+					_, err = nonGatewayRouteHandler.smClient.SubmarinerV1().NonGatewayRoutes(endpoint.Namespace).Create(context.TODO(),
+						nonGatewayRouteHandler.newNonGatewayRoute(endpoint), metav1.CreateOptions{})
+
+					return errors.Wrapf(err, "error updating nonGatewayRoute %q", endpoint.Name)
+				}
+			}
```

Comment:
Same as above re: `util.CreateOrUpdate`.

Reply:
+1

Reply:
when I try to pass  `h.smClient.SubmarinerV1().NonGatewayRoutes(endpoint.Namespace)` util.CreateOrUpdate it is giving me error updateStatus is missing.

Reply:
Can you provide the code you used and the exact message received?

Reply:
I tried this one

Reply:
and the error was `Cannot use 'h.smClient.SubmarinerV1().NonGatewayRoutes(endpoint.Namespace)' (type NonGatewayRouteInterface) as the type resource.Interface Type does not implement 'resource.Interface' as some methods are missing: UpdateStatus(ctx context.Context, obj runtime.Object, options metav1.UpdateOptions) (runtime.Object, error)`

Reply:
ok that's a compile error. You need to pass a `resource.Interface` wrapper. You can either create one for the generated client interfaces, as is done for [Gateways here](https://github.com/submariner-io/submariner/blob/d48ecae9a34ee3227d0c449357857a43431b2bf4/pkg/cableengine/syncer/syncer.go#L107), or just use the general K8s controller-runtime API and [associated resource.Interface](https://github.com/submariner-io/admiral/blob/79a20667691f873c34a509935699aa5114bd59d7/pkg/resource/kubernetes.go#L241).